src/runtime_src/tools/xclbinutil/XclBinUtilities.cxx

```diff
@@ -869,6 +869,14 @@ XclBinUtilities::createMemoryBankGrouping(XclBin & xclbin)
       // Transform and group the memories
       transformMemoryBankGroupingCollections(connectivity, groupTopology, groupConnectivity);
 
+      // Perform some DRC checks on the memory grouping and connectivity produced before marging to group connectivity
+      validateMemoryBankGroupEntries((unsigned int) memTopology.size(), groupTopology, groupConnectivity);
+
+      // Merge connectivity information into the group connectivity
+      for (auto & connection : connectivity) {
```

Comment:
So now we have duplicate entries.

Couldn't find it, how did you make sure that the indexes GROUP memory indexes are correct?  Are you assuming that the memory indexes are at the start.  If so this causes a hard-coded binding which should NOT be there.

Also, could you refer us to the various use-cases that need this data.  

Again, this looks very dangerous.